### ISSUE TITLE
compilation fails on MacPorts builders for Mac OS 10.6, 10.6.x86_64, 10.7, 10.8 and 10.9

### ISSUE THREAD

**Issue by pclayton** [polyml/polyml#166]
MacPorts builders for Mac OS versions from 10.6 to 10.9 all fail to compile with the same error message.

10.9 fails with the following error during compilation ([log](https://build.macports.org/builders/ports-10.9_x86_64-builder/builds/168794/steps/install-port/logs/stdio), [saved text log](https://github.com/polyml/polyml/files/7641632/10.9-stdio.txt))
```
machoexport.cpp:59:10: fatal error: 'mach-o/arm64/reloc.h' file not found
#include <mach-o/arm64/reloc.h>
         ^
```

**Comment by dcjm** [comment]
I've created a MacPorts branch to track this and issue #165.  This should have been fixed in dced042 .

**Comment by pclayton** [comment]
I don't fully understand why this issue occurred because the MacPorts port libmacho-headers should install both header files.  It did on Mac OS 10.10 (from 2014) which is the version I test with.  For some reason, this port on 10.9 and earlier does not provide `/opt/local/include/mach-o/arm64/reloc.h`.  Possibly something about 10.9 and earlier mean it's not possible.

Still, the patch looks good and should resolve this issue, as well as making the code more robust.  I've checked that there is no regression on Mac OS 10.10.
